noidea/git.py

```
@@ -54,6 +62,50 @@ def get_branch_name() -> str:
+
+def get_recent_commits(count: int) -> list[Commit]:
+    """Return up to ``count`` recent non-merge commits as Commit objects (subject + body).
+
+    Best-effort like the other wrappers: any failure (shallow clone, git missing, parse
+    surprise) returns an empty list rather than raising, so the commit hook never breaks.
+    """
+    assert isinstance(count, int) and count > 0, "count must be a positive integer"
+    # check=False + catching FileNotFoundError: a degraded result must fall back to no style
+    # profile, never abort a commit. git missing entirely is just another empty result.
+    try:
+        result = subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)